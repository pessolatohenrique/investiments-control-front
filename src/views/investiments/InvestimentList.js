import React from "react";
import moment from "moment";
import {
  Grid,
  Card,
  CardContent,
  Typography,
  List,
  ListItem,
  ListItemText,
} from "@mui/material";
import InvestimentFactory from "../../factories/InvestimentFactory";

export function InvestimentList({ investiments }) {
  return (
    <Grid container spacing={2}>
      {[...investiments].map((row) => {
        const investimentModel = new InvestimentFactory();
        const investimentCreated = investimentModel.create(row?.type);

        return (
          <Grid item xs={4} key={row.id}>
            <Card data-testid="card-item">
              <CardContent sx={{ minHeight: 265 }}>
                <Typography variant="h6" component="div" color="text.secondary">
                  {row?.description}
                </Typography>
                <List variant="body2">
                  {investimentCreated.renderDetail(row)}
                </List>
              </CardContent>
            </Card>
          </Grid>
        );
      })}
    </Grid>
  );
}
