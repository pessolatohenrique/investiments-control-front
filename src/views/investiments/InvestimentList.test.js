import moment from "moment";
import { render, screen } from "@testing-library/react";
import { InvestimentList } from "./InvestimentList";

const investiments = [
  {
    _id: "6375618ec9e12ca82d98afa7",
    userId: "630ff7a767ae1e8f6a17631b",
    type: "CHECKING_ACCOUNT",
    category: "Conta corrente",
    description: "Conta corrente",
    platform: "ItaÃº",
    net_value: 5000,
    __v: 0,
    profit: 0,
  },
  {
    _id: "637561adc9e12ca82d98afaa",
    userId: "630ff7a767ae1e8f6a17631b",
    goalId: "63507dcba3f4aa14eb41e0ec",
    type: "FIXED_INCOME",
    category: "Tesouro Direto",
    description: "IPCA+ 2026",
    indexer: {
      name: "IPCA",
      contracted_rate: 5.59,
      _id: "63979a5cb3871063914d5c33",
    },
    platform: "Clear",
    final_date: "2022-12-13T00:00:00.000Z",
    monthly_profitability: 0.95,
    invested_amount: 3000,
    expected_net_value: 5000,
    __v: 0,
    net_value: null,
    has_redeemed: false,
    expected_profit: 2000,
  },
];

describe("Goal List", () => {
  it("should generate snapshot", () => {
    const { container } = render(
      <InvestimentList investiments={investiments} />
    );
    expect(container).toMatchSnapshot();
  });

  it("should generate table", () => {
    render(<InvestimentList investiments={investiments} />);
    const rowTable = screen.getAllByTestId("card-item");
    expect(rowTable.length).toBe(2);
  });
});
