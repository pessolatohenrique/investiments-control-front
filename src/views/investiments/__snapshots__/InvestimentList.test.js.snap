// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Goal List should generate snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-mhc70k-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ig64wq-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-r5ajmp-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            Conta corrente
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              variant="body2"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Tipo: 
                    </strong>
                    CHECKING_ACCOUNT
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Valor líquido: 
                    </strong>
                    R$ 
                    5000
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Categoria: 
                    </strong>
                    Conta corrente
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Plataforma: 
                    </strong>
                    Itaú
                  </p>
                </div>
              </li>
            </ul>
          </ul>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Resgatar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Editar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Excluir
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-4 css-gj1fbr-MuiGrid-root"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-14g29go-MuiPaper-root-MuiCard-root"
        data-testid="card-item"
      >
        <div
          class="MuiCardContent-root css-r5ajmp-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h6 css-1jgmn5n-MuiTypography-root"
          >
            IPCA+ 2026
          </div>
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
            variant="body2"
          >
            <ul
              class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              variant="body2"
            >
              <li
                class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
              >
                <div
                  class="MuiListItemText-root css-tlelie-MuiListItemText-root"
                >
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Tipo: 
                    </strong>
                    FIXED_INCOME
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Indexador: 
                    </strong>
                    IPCA
                     + 
                    5.59
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Rentabilidade mensal: 
                    </strong>
                    0.95
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Vencimento em: 
                    </strong>
                    12/12/2022
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Valor esperado: 
                    </strong>
                    R$ 
                    5000
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Valor investido: 
                    </strong>
                    R$ 
                    3000
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Categoria: 
                    </strong>
                    Tesouro Direto
                  </p>
                  <p
                    class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                  >
                    <strong>
                      Plataforma: 
                    </strong>
                    Clear
                  </p>
                </div>
              </li>
            </ul>
          </ul>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Resgatar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Editar
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-fullWidth MuiButtonBase-root  css-1evprfk-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Excluir
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
